wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/registry/ServerInitializerImpl.java

```diff
@@ -104,15 +129,46 @@ protected void registerCallbacks(LanguageServer server, LanguageServerLauncher l
         }
 
         if (launcher instanceof ServerInitializerObserver) {
-            addObserver((ServerInitializerObserver) launcher);
+            addObserver((ServerInitializerObserver)launcher);
         }
     }
 
     private InitializeParams prepareInitializeParams(String projectPath) {
         InitializeParams initializeParams = new InitializeParams();
         initializeParams.setProcessId(PROCESS_ID);
         initializeParams.setRootPath(LanguageServiceUtils.removeUriScheme(projectPath));
-        initializeParams.setCapabilities(new ClientCapabilities());
+        initializeParams.setRootUri(projectPath);
+
+        if (CLIENT_CAPABILITIES == null) {
+            CLIENT_CAPABILITIES = new ClientCapabilities();
+            WorkspaceClientCapabilities workspace = new WorkspaceClientCapabilities();
+            workspace.setApplyEdit(false); //Change when support added
```

Comment:
I believe we can omit anything that is false. I find it confusing to set stuff that is default anyway.